### PR TITLE
SNOW-853914 Add distributed fetch example

### PR DESCRIPTION
**Pull request by sfc-gh-pfus**
### Description
Please explain the changes you made here.

### Checklist
- [ ] Code compiles correctly
- [ ] Run ``make fmt`` to fix inconsistent formats
- [ ] Run ``make lint`` to get lint errors and fix all of them
- [ ] Created tests which fail without the change (if possible)
- [ ] All tests passing
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

**Review by sfc-gh-igarish — COMMENTED** [review]
LGTM. But do we have multi thread like in JDBC to fetch chunks in parallel? Otherwise this will looks like a large resultset fetch.

**Comment by sfc-gh-pfus** [comment]
> LGTM. But do we have multi thread like in JDBC to fetch chunks in parallel? Otherwise this will looks like a large resultset fetch.

@sfc-gh-igarish yes, we have. This is what I meant [here](https://github.com/snowflakedb/gosnowflake/pull/840/files#diff-c521a6b8675547f03ee40c82b33a3db642436d86ac219ab2e863510340b8585dR70). If you observe logs, you can notice that each batch is decoded in separated goroutine.

**Review by sfc-gh-dprzybysz — APPROVED** [review]

**Review by sfc-gh-pbulawa — APPROVED** [review]

**Review by sfc-gh-pmotacki — APPROVED** [review]

File: cmd/distributedfetch/distributedfetch.go [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+		fmt.Printf("ERROR: %v\n", rows.Err())
+		return
+	}
+
+	sf.GetLogger().SetLogLevel("error")
+}
+
+func printSummary() {
+	fmt.Printf(`
+		Please observe as follows.
```
Comment (sfc-gh-dheyman-1) [inline cmd/distributedfetch/distributedfetch.go:68]: This sentence is obsolete. Additionally I would inform a reader about what are the configuration possibilities here (batches size etc.). Also i think this is more of a exampleDescription and not a summary.

**Review by sfc-gh-dheyman-1 — APPROVED** [review]

**Review by sfc-gh-igarish — APPROVED** [review]
LGTM

File: cmd/distributedfetch/distributedfetch.go (hunk 1 above)
Comment (sfc-gh-pfus) [reply cmd/distributedfetch/distributedfetch.go]: 1. Removed first line.
2. Renamed function.
3. I don't think that we should add configuration possibilities. We haven't done that in other examples. Also, this is summary, not configuration instructions :)

Comment (sfc-gh-dheyman-1) [reply cmd/distributedfetch/distributedfetch.go]: Agreed, LGTM

**Pull request merged**